src/vehicle_use.cpp

```diff
@@ -1582,6 +1582,97 @@ void vehicle::use_monster_capture( int part, map */*here*/, const tripoint_bub_m
     invalidate_mass();
 }
 
+void vehicle::use_tiedown_furniture( int part, map *here, const tripoint_bub_ms & )
+{
+    if( !here ) {
+        // should never happen!
+        debugmsg( "vehicle::use_tiedown_furniture() called without a map pointer, please report this error" );
+        return;
+    }
+
+    if( parts[part].is_broken() || parts[part].removed ) {
+        return;
+    }
+
+    const std::string item_var_string = "tied_down_furniture";
+
+    // Why do we construct a copy here?
+    item base_copy = item( parts[part].get_base() );
+
+    if( base_copy.has_var( item_var_string ) ) {
+
+        furn_str_id tied_down_furniture( base_copy.get_var( item_var_string ) );
+        if( !tied_down_furniture.is_valid() ) {
+            debugmsg( "Invalid stored furniture %s", base_copy.get_var( item_var_string ) );
+            base_copy.erase_var( item_var_string ); // to prevent bricking the item
```

Comment:
Because you're operating on a copy, you don't actually remove it from the vehicle part.

Reply:
We set our copy as the base part later on via ` parts[part].set_base( std::move( base_copy ) )` which handles it.  This is actually cargo culted from the use_monster_capture() function right above it. I'm very much unsure of why it does it that way (possibly because they call an iuse?), hence the comment. 

I was imagining there might be some reason not to modify the base part directly, but I did not investigate very far.

(Just wanted to reply to this one for the moment - gotta go through the rest later)

Reply:
Except in this case, where you return from the function without doing that. Sorry, my comment doesn't cover the return statement.

Reply:
...Right! Good catch.